File: Documents/gestion-stock/cpp-version/src/models/Categorie.cpp
#include "../../include/models/Categorie.hpp"

Categorie::Categorie() : id(""), nom(""), description("") {}

Categorie::Categorie(const std::string& id, const std::string& nom, const std::string& description)
    : id(id), nom(nom), description(description) {}

std::string Categorie::getId() const { return id; }
std::string Categorie::getNom() const { return nom; }
std::string Categorie::getDescription() const { return description; }

void Categorie::setNom(const std::string& nom) { this->nom = nom; }
void Categorie::setDescription(const std::string& description) { this->description = description; }

void Categorie::afficher() const {
    std::cout << "ID: " << id << " | Nom: " << nom << " | Desc: " << description << std::endl;
}

File: Documents/gestion-stock/web-version/src/models/Categorie.js
class Categorie {
    constructor(id, nom, description) {
        this.id = id;
        this.nom = nom;
        this.description = description;
    }

    toJSON() {
        return {
            id: this.id,
            nom: this.nom,
            description: this.description
        };
    }
}

module.exports = Categorie;